### DISCUSSION TITLE
config.test.ts 这个配置文件怎么弄才会生效呢？

### DISCUSSION THREAD

**Discussion by peter064226** [umijs/umi#10448] [Q&A]
根据官方文档https://umijs.org/docs/guides/env-variables#umi_env中配置了config.test.ts用于build到测试环境，可是运行"build:test": "cross-env NODE_ENV=test max build"之后读取的是config.prod.ts里的内容，请问如何才会配置才能使其读取config.test.ts里的内容呢？

**Comment by fz6m** [comment]
`NODE_ENV` 是不能改的，可以通过改 `UMI_ENV` 来细分其他的环境，详见 https://umijs.org/docs/guides/env-variables#umi_env 。

**Comment by zlllya** [comment]
build走的是`prod`环境，所以只会加载prod环境的配置文件，因为你配置里面有`config.prod.ts`，按照加载顺序，不想指定的`UMI_ENV`被覆盖，那配置文件应该使用`config.${dev | prod | test}.${UMI_ENV}.ts`的配置文件
